substrate/runtime/contract/src/gas.rs

```diff
@@ -127,6 +137,14 @@ pub fn buy_gas<T: Trait>(
 	transactor: &T::AccountId,
 	gas_limit: T::Gas,
 ) -> Result<GasMeter<T>, &'static str> {
+	// Check if the specified amount of gas is available in the current block.
+	// This cannot underflow since `gas_spent` is never greater than `block_gas_limit`.
+	let gas_available = <Module<T>>::block_gas_limit() - <Module<T>>::gas_spent();
```

Comment:
Why not store `gas_left` instead of the two?

Reply:
You mean instead of `block_gas_limit` and `gas_spent` make just once `gas_left`? 
Hm, I don't see how that could work. At the end of the block we need to replenish `gas_left` to the initial per block limit. We can hardcode it or... add `block_gas_limit` storage variable : )

By the way, I considered using `block_gas_limit` + `gas_left` combo. The reason why I chose `gas_spent` is because you can reset it to zero at the finalization stage. With `gas_left` you need to replenish it before executing the first transaction in the block and it is not obvious to me how to make it cleanly.